### ISSUE TITLE
"no vessel currently matching parameters"

### ISSUE THREAD

**Issue by maekern** [jrossignol/ContractConfigurator#529]
Any contract that I take which is tied to a specific vessel becomes uncompletable once I switch to another vessel, or the space center. On switching back to the target vessel, the objective list says "no vessel currently matching parameters", so no further events involving that vessel are counted towards the mission goals.

[persistent.sfs.txt](https://github.com/jrossignol/ContractConfigurator/files/312691/persistent.sfs.txt)
[output_log.txt](https://github.com/jrossignol/ContractConfigurator/files/312692/output_log.txt)

On the forum I mentioned this seeming related to docking, but I have confirmed that it still happens with a one-piece vessel with no docking involved.

**Comment by maekern** [comment]
Whoops, I apologize for nor giving enough information. The mission in question is the "space camp" mission. The vessel that I launched with the required kerbals is:
pid = b0758f6c1d184a5e9e490a4aeb2c3e76
name = Curious Zeus Akademy

In the save file mission section, I am also seeing references to this vessel, which should not be involved:
pid = 2ebad2b02c7b437295696e2a43a92fbf
name = Curious Zeus Akademy Probe

That is the abandoned final stage of the space camp station, that is still in orbit. I did make sure that the root part of the ship in the VAB is a part that remained on the station part of the vessel, not on that discarded stage.
